### PR TITLE
Don't try hard to compute alignment of unbacked expressions

### PR DESCRIPTION
**Pull request by ezyang**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #132062
* #132061
* #132060
* __->__ #131649

Signed-off-by: Edward Z. Yang <ezyang@meta.com>

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @ColinPeppler @amjames @desertfire @chauhang

### CODE REVIEW

File: torch/_inductor/sizevars.py [hunk 1]
```diff
@@ -356,6 +356,8 @@ def statically_known_multiple_of(
         """
         Return a bool indicating if it is sound to optimize for the numerator being a multiple of the denominator.
         """
+        if free_unbacked_symbols(numerator) or free_unbacked_symbols(denominator):
```
Comment (bdhirsh) [inline torch/_inductor/sizevars.py:359]: i'm a little worried about how `torch._dynamo.decorators.mark_unbacked(x, 0)` will play with this, but I think the answer is that with that API we still provide a hint, and so we wouldn't actually end up with any free unbacked symbols

Comment (eellison) [inline torch/_inductor/sizevars.py:359]: What happens when you run this currently ? I would expect this to just return False if we can't prove it.

 also, for non-view, fresh allocations, we should always be able to guarantee alignment. kind of similar to donated buffers.

File: test/inductor/test_torchinductor_dynamic_shapes.py [hunk 2]
```diff
@@ -912,6 +912,18 @@ def f(x):
...
         f(torch.tensor([5], device=device))
 
+    @torch._dynamo.config.patch(capture_scalar_outputs=True)
+    def test_symint_sum_list(self, device):
+        @torch.compile()
+        def f(xt):
+            xs = xt.tolist()
+            for x in xs:
+                torch._check_is_size(x)
+            y = sum(xs)
```
Comment (bdhirsh) [inline test/inductor/test_torchinductor_dynamic_shapes.py:922]: how does inductor lowering the `sum(List[unbacked symint])` actually end up testing the `statically_known_multiple_of` code?

(i can't think of a case where inductor will be dividing two unbacks symints given this repro)

File: torch/_inductor/sizevars.py (hunk 1 above)
Comment (ezyang) [reply torch/_inductor/sizevars.py:359]: It just returns False, the problem is that it's very expensive to figure out that we're unable to figure it out

File: test/inductor/test_torchinductor_dynamic_shapes.py (hunk 2 above)
Comment (ezyang) [reply test/inductor/test_torchinductor_dynamic_shapes.py:922]: It's not the sum, it's actually the torch.empty call lol

File: torch/_inductor/sizevars.py (hunk 1 above)
Comment (ezyang) [reply torch/_inductor/sizevars.py:359]: It's OK. We will simply assume that it is not aligned!

**Review by bdhirsh — APPROVED** [review]

**Pull request closed without merging**